picard/util/checkupdate.py

```diff
@@ -0,0 +1,177 @@
+            self._display_results()
+        else:
+            # Gets list of releases from specified website api.
+            self._query_available_updates(callback=callback)
+
+    @property
+    def available_versions(self):
+        '''Provide a list of the latest version tuples for each update type.'''
+        return self._available_versions
+
+    def _query_available_updates(self, callback=None):
+        '''Gets list of releases from specified website api.'''
+        log.debug("Getting release information from {host_url}".format(host_url=PLUGINS_API['host'],))
+        self.tagger.webservice.get(
+            PLUGINS_API['host'],
+            PLUGINS_API['port'],
+            PLUGINS_API['endpoint']['releases'],
+            partial(self._releases_json_loaded, callback=callback),
+            parse_response_type=None,
+            priority=True,
+            important=True
+        )
+
+    def _releases_json_loaded(self, response, reply, error, callback=None):
+        '''Processes response from specified website api query.'''
+        if error:
+            log.error(_("Error loading releases list: {error_message}").format(error_message=reply.errorString(),))
+            if self._show_always:
+                QMessageBox.information(
+                    self._parent,
+                    _("Picard Update"),
+                    _("Unable to retrieve the latest version information from the website.\n(https://{url}{endpoint})").format(
+                        url=PLUGINS_API['host'],
+                        endpoint=PLUGINS_API['endpoint']['releases'],
+                    ),
+                    QMessageBox.Ok, QMessageBox.Ok)
+        else:
+            self._available_versions = load_json(response)['versions']
+            for key in self._available_versions:
+                log.debug("Version key '{version_key}' --> {version_information}".format(
+                    version_key=key, version_information=self._available_versions[key],))
+            self._display_results()
+        if callback:
+            callback(not error)
+
+    def _display_results(self):
+        # Display results to user.
+        key = ''
+        high_version = PICARD_VERSION
+        for test_key in PROGRAM_UPDATE_LEVELS:
+            test_version = self._available_versions[PROGRAM_UPDATE_LEVELS[test_key]['name']]['version']
+            if self._update_level >= test_key and  compare_version_tuples(high_version, test_version) > 0:
+                key = PROGRAM_UPDATE_LEVELS[test_key]['name']
+                high_version = test_version
+        if key:
+            if QMessageBox.information(
+                self._parent,
+                _("Picard Update"),
+                _("A new version of Picard is available.\n\n"
+                  "Old version: {picard_old_version}\n"
+                  "New version: {picard_new_version}\n\n"
+                  "Would you like to download the new version?").format(
```

Comment:
I'm a bit bothered by this, because if Picard was installed through a package manager it is a bad idea to suggest the user to download it.
I wonder if we should provide an easy way for packagers to disable this feature...

What do you think guys?

Reply:
We could change this to something like, "Visit the website for more information?", and just link to `https://picard.musicbrainz.org/" or "https://picard.musicbrainz.org/changelog/" for stable releases (set in the PW configuration).

Another option is to leave out the link option entirely, and just have this as a notification only.  If the user wants to update, they'll find a way.

Reply:
Linking to the Changelog with some `View recent changes` title could do.

Reply:
I actually would leave it as is for the case that we show update information. As a user I expect the software that has built-in update checks to actually direct me to the download.

But @zas' concern about the need for packagers to disable this is valid. As a packager I ideally would want to have an option when running the install via setup.py to completely disable this feature (no visible option, no update checking performed).

But crippling the displayed message doesn't help both cases: Distribution users still get bothered with update information even though they depend on their distribution to update the packages and have no easy way to update themselves, and users who installed manually get an update information where they then have to hunt for the download page.

Reply:
Having said this IMHO we should probably merge this PR and open a separate PR for disabling the update check. I am also not really sure how to technically best solve this.

Reply:
I'm not very familiar with what happens when installing from a packager, but if there is (or could be) a specific file (e.g.: PACKAGER.txt) added to the program directory, we should be able to check for that and disable the update checking and hide the options in the toolbar and settings.  Would that work?

I've already coded the changes based on Picard finding a `PACKAGER.txt` file, and can include them here if you like.  Then the other PR could just sort out the `INSTALLED_BY_PACKAGER` test if we wanted something different.